### PR TITLE
Fix gcc-11 warnings: signed/unsigned integer comparison, unused variables

### PR DESCRIPTION
**Pull request by cary-ilm**

### CODE REVIEW

**Comment by cary-ilm** [comment]
@kdt3rd, @meshula, @peterhillman, @lgritz, these seem fairly benign, but I'd still like some other eyes on it, can someone look it over?

File: src/test/OpenEXRTest/testDwaLookups.cpp [hunk 1]
```diff
@@ -228,8 +228,7 @@ testNoop ()
 
     for (int i = 0; i < 65536; ++i)
     {
-        unsigned short src = (unsigned short) i;
-        assert (src == OPENEXR_IMF_INTERNAL_NAMESPACE::dwaCompressorNoOp[i]);
+        assert ((unsigned short) i == OPENEXR_IMF_INTERNAL_NAMESPACE::dwaCompressorNoOp[i]);
```
Comment (meshula) [inline src/test/OpenEXRTest/testDwaLookups.cpp]: Should probably just make the loop

for (unsigned short i = 0; ...)

then there's no casting at all

File: src/test/OpenEXRCoreTest/general_attr.cpp [hunk 2]
```diff
@@ -53,12 +53,12 @@ failable_malloc (size_t bytes)
     return malloc (bytes);
 }
 
-static void
-failable_free (void* p)
```
Comment (meshula) [inline src/test/OpenEXRCoreTest/general_attr.cpp]: down below we assign failable_alloc within cinit.
I'm pretty sure the intent was failable_free be correspondingly assigned within cinit as well.

**Review by meshula — CHANGES_REQUESTED** [review]
I think failable_free was meant to be assigned ~ could we try that? Hopefully it doesn't introduce a new error.

File: src/test/OpenEXRTest/testDwaLookups.cpp (hunk 1 above)
Comment (cary-ilm) [reply src/test/OpenEXRTest/testDwaLookups.cpp]: fixed

File: src/test/OpenEXRCoreTest/general_attr.cpp (hunk 2 above)
Comment (cary-ilm) [reply src/test/OpenEXRCoreTest/general_attr.cpp]: Adding the assignment to `free_fn`:
```
static exr_context_t
createDummyFile (const char* test)
{
    exr_context_t             f     = NULL;
    exr_context_initializer_t cinit = EXR_DEFAULT_CONTEXT_INITIALIZER;

    // we won't actually write to this and so don't need a proper
    // stream but need a writable context to test with.
    cinit.write_fn = dummy_write;
    cinit.alloc_fn = failable_malloc;
    cinit.free_fn = failable_free;
```
causes `OpenEXRCore.testAttrChlists` and `OpenEXRCore.testAttrLists` to fail, so something needs some further attention. I'll leave the free function in for now.

Comment (meshula) [reply src/test/OpenEXRCoreTest/general_attr.cpp]: Gotcha, can we add 

```cpp
// adding failable_malloc currently causes OpenEXRCore.testAttrChlists and OpenEXRCore.testAttrLists to fail.
// commented out until we've investigated.
//cinit.free_fn = failable_free;
```

With that I think we're good to land this

**Review by meshula — APPROVED** [review]

**Pull request merged**